doc/sphinx/source/howto/execute.rst

```diff
@@ -34,42 +34,37 @@ Define the computation
+outputs. While a :term:`bridge` can programmatically construct the graph 
 from a framework's representation of the computation, graph construction can be 
 somewhat more tedious for users. To a user, who is usually interested in 
 specific nodes (vertices) or edges of a computation that reveal "what is 
 happening where", it can be helpful to think of a computation as a zoomed-out 
-and *stateless* dataflow graph where all of the nodes are well-defined tensor 
-operations and all of the edges denote use of an output from one operation as 
-an input for another operation.
-
-.. TODO
-
-.. image for representing nodes and edges of (a+b)*c
-
+and *stateless* :term:`data-flow graph` where all of the nodes are well-defined 
+tensor operations and all of the edges denote use of an output from one 
+operation as an input for another operation.
 
 Most of the public portion of the nGraph API is in the ``ngraph`` namespace, so 
 we will omit the namespace. Use of namespaces other than ``std`` will be 
 namespaces in ``ngraph``. For example, the ``op::Add`` is assumed to refer to 
-``ngraph::op::Add``.
-
-A computation's graph is constructed from ops; each is a member of a subclass of 
-``op::Op``, which, in turn, is a subclass of ``Node``. Not all graphs are 
-computation, but all graphs are composed entirely of instances of ``Node``.  
-Computation graphs contain only ``op::Op`` nodes.
+``ngraph::op::Add``. A computation's graph is constructed from ops; each is a 
+member of a subclass of ``op::Op``, which, in turn, is a subclass of ``Node``. 
+Not all graphs are computation, but all graphs are composed entirely of 
+instances of ``Node``.  Computation graphs contain only ``op::Op`` nodes.
 
 We mostly use :term:`shared pointers<shared pointer>` for nodes, i.e.
-``std::shared_ptr<Node>`` so that they will be automatically
-deallocated when they are no longer needed. A brief summary of shared
-pointers is given in the glossary.
+``std::shared_ptr<Node>`` so that they will be automatically deallocated when 
+they are no longer needed. More detail on shared pointers is given in the 
+glossary.
 
 Every node has zero or more *inputs*, zero or more *outputs*, and zero or more 
-*attributes*.  The specifics for each ``type`` permitted on a core ``Op``-specific 
-basis can be discovered in our :doc:`../ops/index` docs. For our 
-purpose to :ref:`define a computation <define_cmp>`, nodes should be thought of 
-as essentially immutable; that is, when constructing a node, we need to supply 
-all of its inputs. We get this process started with ops that have no inputs, 
-since any op with no inputs is going to first need some inputs.
+*attributes*. 
+
+The specifics for each ``type`` permitted on a core ``Op``-specific basis can be 
+discovered in our :doc:`../ops/index` docs. For our purpose to 
+:ref:`define a computation <define_cmp>`, nodes should be thought of as essentially 
+immutable; that is, when constructing a node, we need to supply all of its 
```

Comment:
We need to figure out how to reconcile this immutability with patterns/splicing, whose sole purpose is modifying the graph. For now this is okay; it's a good way to think of things while building the graph.